### ISSUE TITLE
Allow set notation in `allow-newer`

### ISSUE THREAD

**Issue by andreasabel** [haskell/cabal#7786]
```
$ cabal build --allow-newer='aeson:{base,ghc-prim}'
Error: cabal: invalid argument to option `--allow-newer': Cannot parse the
list of packages: aeson:{base,ghc-prim}
```
Would be nice if this would be expanded in the obvious way.

In general, in `--allow-newer=P:D`, `P:D` could mean the cartesian product of sets `P` and `D`.

Related: #7740

**Comment by josephcsible** [comment]
If you omit the single quotes, won't this already do what you want due to the shell's expansion of curly braces?

**Comment by andreasabel** [comment]
Indeed, at least in bash: https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Brace-Expansion
Learned something new!
But this wouldn't work then if I call cabal directly, or via powershell, or...

**Comment by philderbeast** [comment]
This set notation could also be used in project files.

**Comment by ulysses4ever** [comment]
For a future volunteer taking up this ticket: does anyone have a place in cabal syntax in mind that already supports set syntax?
